fvm/transactionAccountFrozen.go

```diff
@@ -0,0 +1,56 @@
+package fvm
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type TransactionAccountFrozenChecker struct{}
+
+func NewTransactionAccountFrozenChecker() *TransactionAccountFrozenChecker {
+	return &TransactionAccountFrozenChecker{}
+}
+
+func (c *TransactionAccountFrozenChecker) Process(
+	_ *VirtualMachine,
+	_ Context,
+	proc *TransactionProcedure,
+	st *state.State,
+) error {
+
+	return c.checkAccountNotFrozen(proc.Transaction, st)
+}
+
+func (c *TransactionAccountFrozenChecker) checkAccountNotFrozen(
+	tx *flow.TransactionBody,
+	st *state.State,
+) error {
+	accounts := state.NewAccounts(st)
+
+	errIfFrozen := func(address flow.Address) error {
+		frozen, err := accounts.GetAccountFrozen(address)
+		if err != nil {
+			return fmt.Errorf("cannot check acount free status: %w", err)
+		}
+		if frozen {
+			return &AccountFrozenError{Address: address}
+		}
+		return nil
+	}
```

Comment:
This seems to be a copy of `hostEnv.isAccountFrozen`, https://github.com/onflow/flow-go/pull/329/files#diff-b4e3be012ea5724c7d1faf9e1ffa2780351d5c876037ccd25ee33704d587e676R587. Maybe use that function (might have to be refactored out of `hostEnv`, so that it only on `state.Accounts`)